neuro_san/internals/graph/persistence/registry_manifest_restorer.py

```diff
@@ -118,23 +125,27 @@ def restore_from_files(self, file_references: Sequence[str]) -> Dict[str, AgentN
                 raise FileNotFoundError(message)
 
             for key, value in one_manifest.items():
+                if not bool(value):
+                    # Fast out
+                    continue
 
+                # Key here is an agent name in a form that we chose,
+                # and we'll need to use an agent mapper to get to this agent definition file.
                 # Keys sometimes come with quotes.
                 use_key: str = key.replace(r'"', "")
-
-                if not bool(value):
-                    continue
+                use_key = use_key.strip()
+                agent_filepath: str = self.agent_mapper.agent_name_to_filepath(use_key)
 
                 file_of_class = FileOfClass(manifest_file)
                 manifest_dir: str = file_of_class.get_basis()
-                registry_restorer = AgentNetworkRestorer(manifest_dir)
+                registry_restorer = AgentNetworkRestorer(registry_dir=manifest_dir, agent_mapper=self.agent_mapper)
                 try:
-                    agent_network: AgentNetwork = registry_restorer.restore(file_reference=use_key)
+                    agent_network: AgentNetwork = registry_restorer.restore(file_reference=agent_filepath)
                 except FileNotFoundError as exc:
                     self.logger.error("Failed to restore registry item %s - %s", use_key, str(exc))
                     agent_network = None
                 if agent_network is not None:
-                    network_name: str = Path(use_key).stem
+                    network_name: str = self.agent_mapper.filepath_to_agent_network_name(agent_filepath)
                     agent_networks[network_name] = agent_network
```

Comment:
From agent definition file location, get the agent network name that will identify it to the service.